bin/po2swift.js

```diff
@@ -1,47 +1,74 @@
-	fs = require( 'fs' );
+	fs = require( 'fs' ),
+	path = require( 'path' );
 
-function po2Swift( poInput ) {
-	const po = gettextParser.po.parse( poInput );
-	const translations = po.translations[ '' ];
-	const swiftStringsMap = Object.values( translations ).reduce( ( result, translation ) => {
-		if ( ! translation.msgid ) {
+function po2Swift( strings ) {
+	const swiftStringsMap = strings.reduce(
+		( result, { string, comments } ) => {
+			if ( ! string ) {
+				return result;
+			}
+			const encodedValue = JSON.stringify( string );
+			const encodedComment = JSON.stringify( comments || '' );
+			result[
+				string
+			] = `_ = NSLocalizedString(${ encodedValue }, comment: ${ encodedComment })`;
 			return result;
-		}
-		const encodedValue = JSON.stringify( translation.msgid );
-		const encodedComment = JSON.stringify( translation.comments.extracted || '' );
-		result[ translation.msgid ] = `_ = NSLocalizedString(${ encodedValue }, comment: ${ encodedComment })`;
-		return result;
-	}, {} );
+		},
+		{}
+	);
 	const swiftStringsSortedList = Object.entries( swiftStringsMap )
 		.sort( ( left, right ) => left[ 0 ].localeCompare( right[ 0 ] ) )
 		.map( ( entry ) => entry[ 1 ] );
-	return `import Foundation\n\nprivate func dummy() {\n    ${ swiftStringsSortedList.join( '\n    ' ) }\n`;
+	return `import Foundation\n\nprivate func dummy() {\n    ${ swiftStringsSortedList.join(
+		'\n    '
+	) }\n`;
 }
 
+// Returns used strings of a specified domain
+const getUsedStrings = ( usedStringsFile ) => {
+	return require( path.resolve( usedStringsFile ) );
+};
+
 if ( require.main === module ) {
-	if ( process.stdin.isTTY ) {
-		const potFileName = process.argv[ 2 ];
-		const destination = process.argv[ 3 ];
-		const potFileContent = fs.readFileSync( potFileName );
-		const swiftOutput = po2Swift( potFileContent, process.argv[ 3 ] );
-		fs.writeFileSync( destination, swiftOutput );
-	} else {
-		let inputData = '';
-		process.stdin.on( 'readable', function() {
-			const chunk = this.read();
-			if ( chunk !== null ) {
-				inputData += chunk;
-			}
-		} );
-		process.stdin.on( 'end', function() {
-			process.stdout.write( po2Swift( inputData ) );
-		} );
+	const destination = process.argv[ 2 ];
+	const usedStringsFile = process.argv[ 3 ];
+
+	const usedStrings = getUsedStrings( usedStringsFile );
+	let flattenUsedStrings = Object.entries( usedStrings ).reduce(
+		( result, [ domain, strings ] ) => {
+			return [ ...result, ...Object.values( strings ) ];
+		},
+		[]
+	);
+	flattenUsedStrings = [ ...new Set( flattenUsedStrings ) ];
+
+	const nativeStringsWithContext = [];
+	const onlyNativeStrings = flattenUsedStrings.filter( ( item ) => {
+		const { platforms, string } = item;
+
+		const isNative =
+			! platforms.includes( 'web' ) && platforms.includes( 'ios' );
+		const hasContext = !! item.context;
+		if ( isNative && hasContext ) {
+			nativeStringsWithContext.push( string );
+		}
+
+		return isNative && ! hasContext;
+	}, [] );
```

Comment:
We only add the localization strings files of strings referenced in `.native.js` and `.ios.js files`, and that doesn't have context.